File: python/merge/Merger_pdf_doc.py
from aspose.pdf import (
    DocSaveOptions,
    Document
)


def pdf_to_doc():
    path_source1 = "../../TestData/test.pdf"
    path_source2 = "../../TestData/Second/test.pdf"

    # read pdf file to Aspose Document
    first_doc = Document(path_source1)
    second_doc = Document(path_source2)

    # create empty pdf document
    output_doc = Document()

    # set less memory usage with unload instead of fast performance
    output_doc.enable_object_unload = True

    for page in first_doc.pages:
        # add page from one document to another directly
        output_doc.pages.add(page)

    for page in second_doc.pages:
        # add page from one document to another directly
        output_doc.pages.add(page)

    opt1 = DocSaveOptions()
    # use doc format
    opt1.format = DocSaveOptions.DocFormat.DOC
    # This mode is fast and good for maximally preserving original look
    opt1.mode = DocSaveOptions.RecognitionMode.TEXTBOX

    # save pdf to Microsoft Word doc format
    output_doc.save("Merger_pdf_doc.doc", opt1)

File: cpp/merge/Merger_pdf_doc.cpp
#include "Aspose.PDF.Cpp\Document.h"
#include "Aspose.PDF.Cpp\Page.h"
#include "Aspose.PDF.Cpp\DocSaveOptions.h"
using namespace System;
using namespace Aspose::Pdf;

void pdf_to_doc()
{
    auto pathSource1 = u"../../TestData/test.pdf";
    auto pathSource2 = u"../../TestData/Second/test.pdf";

    // read pdf file to Aspose Document
    auto firstDoc = MakeObject<Document>(pathSource1);
    auto secondDoc = MakeObject<Document>(pathSource2);

    // create empty pdf document
    auto outputDoc = MakeObject<Document>();

    // set less memory usage with unload instead of fast performance
    outputDoc->EnableObjectUnload = true;

    for (int i = 0; i < firstDoc->get_Pages()->get_Count(); i++)
    {
        auto page = firstDoc->get_Pages()->get_Item(i);
        // add page from one document to another directly
        outputDoc->get_Pages()->Add(page);
    }

    for (int i = 0; i < secondDoc->get_Pages()->get_Count(); i++)
    {
        auto page = secondDoc->get_Pages()->get_Item(i);
        // add page from one document to another directly
        outputDoc->get_Pages()->Add(page);
    }

    var opt1 = MakeObject<DocSaveOptions>();
    // use doc format
    opt1->Format = DocFormat::Doc;
    // This mode is fast and good for maximally preserving original look
    opt1->Mode = RecognitionMode::Textbox;

    outputDoc->Save("test.doc", opt1);
}
